lib/generateClass.js

```diff
@@ -205,6 +213,61 @@ function writeFromJson(def, cw) {
   cw.ln('return inst;');
 }
 
+/**
+ * Generates a JSON serializer for the element
+ * @param {DataElement} def - The definition of the SHR element to generate a serializer for
+ * @param {CodeWriter} cw - The CodeWriter instance to use during generation
+ * @private
+ */
+function writeToJson(def, cw) {
+  // If the element is an Entry, put those fields on the JSON object first
+  if (def.isEntry) {
+    cw.ln(`const inst = this._entryInfo.toJSON();`);
+  } else if (def.identifier.name !== 'EntryType') {
+    const url = `http://standardhealthrecord.org/spec/${def.identifier.namespace.replace('.', '/')}/${def.identifier.name}`;
+    cw.ln(`const inst = { 'shr.base.EntryType': { 'Value' : '${url}' } };`);
+  } else {
+    cw.ln(`const inst = {};`);
+  }
+
+  if (def.value !== undefined) {
+    cw.bl('if (this.value !== undefined)', () => {
+      if (def.value instanceof ChoiceValue) {
+        // Choices get a special treatment
+        cw.ln(`inst['Value'] = typeof this.value.toJSON === 'function' ? this.value.toJSON() : this.value;`);
+      } else if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
+        cw.ln(`inst['Value'] = this.value;`);
+      } else {
+        generateAssignmentIfList(def.value.card, 'Value', 'value', cw);
+      }
+    });
+    
+  }
+
+  for (const field of def.fields) {
+    if (!field.identifier.name instanceof TBD) {
+      generateAssignmentIfList(field.card, field.effectiveIdentifier.fqn, toSymbol(field.identifier.name), cw);
```

Comment:
The second argument should just be `field.identifier.fqn`.  The class generator always preserves property names in subclasses, so even if a subclass constrains a field to a more specific type, we keep the original property name (to make the inheritance clear).

For example, consider an element called `Foo` with a field called `Bar` (I'm getting no creativity points here), and another element called `SonOfFoo` that is based on `Foo` and constrains `Bar` to be a `SonOfBar`.

```js
class Foo {
  // code skipped for brevity
  get bar() { return this._bar; }
}

class SonOfFoo extends Foo {
  // code skipped for brevity
  // If we had the next line, the relationship between the bar getter and sonOfBar getter is unclear:
  // get sonOfBar() { return this._bar; }
  // So we just keep the original name to make it clear it is the same property:
  get bar() { return this._bar; }
}

Reply:
Ha.  I just realized that what I export as properties in the generated class is kind of a moot point, because you're concerned w/ matching the JSON schema, not necessarily the class.  Doh.

But... the overall point still applies, because that's how the JSON Schema works too.  For example, if you look at the generated schema for the `OverrideBasedOnIntegerValueElementEntry` test fixture, the `StringValue` that is constrained to be a `StringValueChild`, still has a property name reflecting the identifier (`StringValue`), not the effectiveIdentifier (`StringValueChild`).
```json
          "properties": {
            "shr.simple.StringValue": {
              "$ref": "#/definitions/StringValueChild"
            }
          },
```